src/inc/til/rle.h

```diff
@@ -866,9 +885,69 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
 
             // TODO GH#10135: Ensure replacements contains no runs with .length == 0.
 
-            rle_scanner scanner{ _runs.begin(), _runs.end() };
-            auto [begin, begin_pos] = scanner.scan(start_index);
-            auto [end, end_pos] = scanner.scan(end_index);
+            // Fast path: replace all
+            if (start_index == 0 && end_index == _total_length)
+            {
+                _runs.clear();
+                _total_length = 0;
+                _runs.insert(_runs.begin(), replacements.begin(), replacements.end());
+                const auto size = replacements.size();
+                const auto mid = size / 2;
+                for (size_t i = 0; i < size; i++)
+                {
+                    if (i == mid)
+                    {
+                        _hint_run = i;
+                        _hint_index = _total_length;
+                    }
+                    _total_length += replacements[i].length;
+                }

```

Comment:
Is calculating the hint necessary here? It would get figured out automatically on the next call, no?